lib/PasswordStrength/PasswordStrength.js

```diff
@@ -2,45 +2,102 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
 import taiPasswordStrength from 'tai-password-strength';
+import TextField from '@folio/stripes-components/lib/TextField';
+import { Row, Col } from '@folio/stripes-components/lib/LayoutGrid';
+import InfoPopover from '@folio/stripes-components/lib/InfoPopover';
+
+import css from './PasswordSrength.css';
 
 class PasswordStrength extends React.Component {
   static propTypes = {
-    password: PropTypes.string.isRequired
+    /**
+     * Properties for column that includes textInput
+     */
+    inputColProps: PropTypes.object,
+    /**
+     * Properties for column that includes password strength meter
+     */
+    passwordMeterColProps: PropTypes.object,
+    /**
+     * Is password strength meter hidden
+     */
+    hidden: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.strengthTester = new taiPasswordStrength.PasswordStrength();
     this.strengthTester.addCommonPasswords(taiPasswordStrength.commonPasswords);
     this.strengthTester.addTrigraphMap(taiPasswordStrength.trigraphs);
-    this.output = 'weak'; // set default password hint to weak
+    this.output = 'veryWeak'; // set default password hint to weak
   }
 
-  componentDidUpdate(prevProps) {
-    // in case password updated and provided
-    if (this.props.password && this.props.password !== prevProps.password) {
-      const password = this.props.password;
-      const result = this.strengthTester.check(password);
+  componentDidUpdate() {
+    const { input: { value } } = this.props;
+    const result = this.strengthTester.check(value);
 
-      if (result.strengthCode.indexOf('WEAK') >= 0) {
+    switch (result.strengthCode) {
+      case 'WEAK':
         this.output = 'weak';
-      } else if (result.strengthCode.indexOf('STRONG') >= 0) {
-        this.output = 'strong';
-      } else {
+        break;
+      case 'REASONABLE':
         this.output = 'reasonable';
-      }
+        break;
+      case 'STRONG':
+        this.output = 'strong';
+        break;
+      case 'VERY_STRONG':
+        this.output = 'veryStrong';
+        break;
+      default:
+        this.output = 'veryWeak';
+        break;
     }
   }
 
   render() {
-    if (!this.props.password) {
-      return null;
-    }
+    const { meta: { valid, dirty }, inputColProps, passwordMeterColProps, hidden } = this.props;
 
     return (
-      <div>
-        <FormattedMessage id={'stripes-smart-components.passwordStrength.' + this.output} />
-      </div>
+      <Row>
+        <Col xs={6} {...inputColProps}>
+          <TextField {...this.props} />
+        </Col>
+        <Col xs={4} {...passwordMeterColProps} aria-live="polite">
+          {valid && !hidden && dirty &&
```

Comment:
Let's have `valid && !hidden && dirty` expression as its own variable for readability purpose.